File: csharp/Agentic.Commerce.Protocol/Models/AllowanceReason.cs
// this file is @generated
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Agentic.Commerce.Protocol
{
    [JsonConverter(typeof(StringEnumConverter))]
    public enum AllowanceReason
    {
        [EnumMember(Value = "one_time")]
        OneTime = 1,
    }
}

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/AllowanceReason.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class AllowanceReason : ToQueryParam {
    @SerialName("one_time") ONE_TIME;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
